core/controllers/android_e2e_config.py

```diff
@@ -0,0 +1,288 @@
+            topic.add_subtopic(1, 'Test Subtopic Title')
+            topic.update_subtopic_url_fragment(1, 'suburl')
+            topic.update_subtopic_thumbnail_filename(1, 'test_svg.svg')
+            topic.update_subtopic_thumbnail_bg_color(1, '#FFFFFF')
+
+            topic.move_skill_id_to_subtopic(None, 1, skill_id)
+
+            subtopic_page = (
+                subtopic_page_domain.SubtopicPage.create_default_subtopic_page(
+                    1, topic_id))
+
+            exp_services.load_demo(python_utils.convert_to_bytes(
+                exp_id))
+            rights_manager.release_ownership_of_exploration(
+                user_services.get_system_user(), python_utils.convert_to_bytes(
+                    exp_id))
+            exp_services.update_exploration(
+                user_id, exp_id, [exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                    'property_name': 'correctness_feedback_enabled',
+                    'new_value': True
+                })], 'Changed correctness_feedback_enabled.')
+            story = story_domain.Story.create_default_story(
+                story_id, 'Android End to End testing', 'Description',
+                topic_id, 'android-end-to-end-testing')
+
+            story_node_dicts = [{
+                'exp_id': exp_id,
+                'title': 'Testing with UI Automator',
+                'description': 'In order to test all android interactions'
+            }]
+
+            def generate_dummy_story_nodes(node_id, exp_id, title, description):
+                """Generates and connects sequential story nodes.
+
+                Args:
+                    node_id: int. The node id.
+                    exp_id: str. The exploration id.
+                    title: str. The title of the story node.
+                    description: str. The description of the story node.
+                """
+
+                story.add_node(
+                    '%s%d' % (story_domain.NODE_ID_PREFIX, node_id),
+                    title)
+                story.update_node_description(
+                    '%s%d' % (story_domain.NODE_ID_PREFIX, node_id),
+                    description)
+                story.update_node_exploration_id(
+                    '%s%d' % (story_domain.NODE_ID_PREFIX, node_id), exp_id)
+
+                story.update_node_thumbnail_filename(
+                    '%s%d' % (story_domain.NODE_ID_PREFIX, node_id),
+                    'test_svg.svg')
+                story.update_node_thumbnail_bg_color(
+                    '%s%d' % (story_domain.NODE_ID_PREFIX, node_id), '#F8BF74')
+                story.update_meta_tag_content('tag')
+                story.update_thumbnail_filename('test_svg.svg')
+                story.update_thumbnail_bg_color(
+                    constants.ALLOWED_THUMBNAIL_BG_COLORS['story'][0])
+
+                if node_id != len(story_node_dicts):
```

Comment:
I'm not sure, I used [admin.py](https://github.com/oppia/oppia/blob/develop/core/controllers/admin.py#L449) as reference for this as well. This was added in https://github.com/oppia/oppia/pull/9473

The function [generate_dummy_story_nodes](https://github.com/oppia/oppia/blob/develop/core/controllers/admin.py#L430) is necessary for adding the exploration to the story.

Reply:
I just found that this isn't neccessary here, this condition is to update only the last node from the `story_node_dicts` and in our case there is only one node.

Shall I remove this whole function and do It without the function call or leave it as we might expect more nodes or explorations in this story.

Reply:
Leaving this may be a issue for the code coverage.
![image](https://user-images.githubusercontent.com/54740946/123555899-2037c180-d7a6-11eb-9bbf-fa31ae703ea5.png)
here the missing line 151 indicates the function call below `if`

Reply:
If this isn't needed, let's remove it.

Reply:
Done